### PR TITLE
[feature][dart] Add support for uniqueItems/sets

### PR DESCRIPTION
**Pull request by kuhnroyal**
* copy `uniqueItems` usage from 2.0 fake spec to 3.0
* add support for sets in parameters, responses and properties

### PR checklist
 
- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ ] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [ ] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [ ] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by kuhnroyal** [comment]
@agilob @josh-burton Please review if you find time

**Review by josh-burton — APPROVED** [review]

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/dartdio/DartDioModelTest.java [hunk 1]
```diff
@@ -204,6 +204,44 @@ public void listPropertyTest() {
         Assert.assertTrue(property2.isContainer);
     }
 
+    @Test(description = "convert a model with set property")
```
Comment (agilob) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/dartdio/DartDioModelTest.java:207]: finally some tests :D

**Comment by agilob** [comment]
LGTM, not sure whether other people maintaining jersey will be happy with changes in spec, but it looks like it works for their generator too.

I probably should start using dart-dio, you're doing great work there mate.

**Review by agilob — APPROVED** [review]

**Comment by kuhnroyal** [comment]
This is probably considered a breaking change without fallback? So it goes to 6.0.0? 😢

**Comment by wing328** [comment]
> This is probably considered a breaking change without fallback? So it goes to 6.0.0? 😢

We did not consider adding set support is a breaking change in other generators (e.g. Ruby, Java, etc) before so I'm ok this change gets into the current master.

**Comment by kuhnroyal** [comment]
@wing328 Sounds good to me, I have removed the draft label. Build failures seem unrelated as far as I can tell.

**Comment by wing328** [comment]
Since there are changes to the test spec, can you please run `./bin/generate-samples.sh` to update the samples for other generators?

**Comment by kuhnroyal** [comment]
I did, thats what caused the changes in TS/Kotlin. I can rebase and do that again, maybe something has changed in the meantime.

**Comment by wing328** [comment]
Can you merge the latest master into your branch and update the samples again?

I just did that and saw some changes to the Ruby and C# samples.

**Comment by kuhnroyal** [comment]
Rebased and regenerated.

**Pull request merged**